components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtService.java

```diff
@@ -57,6 +57,24 @@ TenantSearchResult listTenants(Integer limit, Integer offset, String sortOrder,
      */
     Tenant getTenant(String tenantUniqueIdentifier) throws TenantMgtException;
 
+    /**
+     * Get a specific tenant using tenant id.
+     *
+     * @param domain tenant domain.
+     * @return Tenant
+     * @throws TenantMgtException if getting the tenant fails.
+     */
+    Tenant getTenantByDomain(String domain) throws TenantMgtException;
```

Comment:
Shouldn't we have default methods for these to be backward compatible ? Unless there can be issues for the people who had implemented extensions.. Should these be implemented necessarily ? If so ignore this.

Reply:
This is a newly added interface. No product release is done with that. So this should be fine.